config/plugins/pbh.py

```diff
@@ -0,0 +1,852 @@
+    data = {}
+    if hash_field_list is not None:
+        data["hash_field_list"] = hash_field_list.split(",")
+
+    try:
+        add_entry(db.cfgdb, table, key, data)
+    except Exception as err:
+        exit_with_error("Error: {}".format(err), fg="red")
+
+
+@PBH_HASH.command(name="update")
+@click.argument(
+    "hash-name",
+    nargs=1,
+    required=True,
+)
+@click.option(
+    "--hash-field-list",
+    help="The list of hash fields to apply with this hash",
+)
+@clicommon.pass_db
+def PBH_HASH_update(db, hash_name, hash_field_list):
+    """ Update object in PBH_HASH table """
+
+    if not is_exist_in_db(db, hash_field_list, pbh_hash_field_tbl_name):
+        exit_with_error("Error: invalid value '{}' for '--hash-field-list' option".format(hash_field_list), fg="red")
+
+    table = pbh_hash_tbl_name 
+    key = hash_name
+    data = {}
+    if hash_field_list is not None:
+        data["hash_field_list"] = hash_field_list.split(",")
+
+    try:
+        update_entry(db.cfgdb, table, key, data)
+    except Exception as err:
+        exit_with_error("Error: {}".format(err), fg="red")
+
+
+@PBH_HASH.command(name="delete")
+@click.argument(
+    "hash-name",
+    nargs=1,
+    required=True,
+)
+@clicommon.pass_db
+def PBH_HASH_delete(db, hash_name):
+    """ Delete object from PBH_HASH table """
+
+    table = pbh_hash_tbl_name
+    key = hash_name
+    try:
+        del_entry(db.cfgdb, table, key)
+    except Exception as err:
+        exit_with_error("Error: {}".format(err), fg="red")
+
+
+@PBH.group(
+    name="rule",
+    cls=clicommon.AliasedGroup
+)
+def PBH_RULE():
+    """ Configure PBH rule """
+
+    pass
+
+
+@PBH_RULE.command(name="add")
+@click.argument(
+    "table-name",
+    nargs=1,
+    required=True,
+)
+@click.argument(
+    "rule-name",
+    nargs=1,
+    required=True,
+)
+@click.option(
+    "--priority",
+    help="Configures priority",
+    required=True,
+    type=click.INT,
+)
+@click.option(
+    "--gre-key",
+    help="Configures packet match: GRE key (value/mask)",
+    callback=pbh_re_match_validator,
+)
+@click.option(
+    "--ether-type",
+    help="Configures packet match for this rule: EtherType (IANA Ethertypes)",
+    callback=pbh_re_match_validator,
+)
+@click.option(
+    "--ip-protocol",
+    help="Configures packet match for this rule: IP protocol (IANA Protocol Numbers)",
+    callback=pbh_re_match_validator,
+)
+@click.option(
+    "--ipv6-next-header",
+    help="Configures packet match for this rule: IPv6 Next header (IANA Protocol Numbers)",
+    callback=pbh_re_match_validator,
+)
+@click.option(
+    "--l4-dst-port",
+    help="Configures packet match for this rule: L4 destination port",
+    callback=pbh_re_match_validator,
+)
+@click.option(
+    "--inner-ether-type",
+    help="Configures packet match for this rule: inner EtherType (IANA Ethertypes)",
+    callback=pbh_re_match_validator,
+)
+@click.option(
+    "--hash",
+    required=True,
+    help="Configures the hash to apply",
```

Comment:
@vadymhlushko-mlnx please align the description with update subgroup - use YANG model as reference

Reply:
@nazariig fixed